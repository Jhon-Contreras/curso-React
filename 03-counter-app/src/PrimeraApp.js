// Los componentes, se escriban con UpperCamelCase 

// para usar Fragment importar desde react 
// import { Fragment } from "react";

// en caso de error de JSX se necesitaría importar react 


import PropTypes from 'prop-types';

// Functional components

// props con destructuracion 
const PrimeraApp = ( { saludo } ) => {

    // const saludo = 'Hola mundo';
    // const saludoObjeto = {
    //     nombre : 'JHhon',
    //     edad: 2
    // };

    return (
       <>
            {/* <pre> { JSON.stringify(saludoObjeto, null, 3) } </pre> */}
            <h1> { saludo } </h1>
            <p>Soy un parrafo</p>
       </>
        // componente con Fragment (necesita importar libreria) 

        // <Fragment>
        // <h1> Hola Mundo desde un Functional Component</h1>
        // <p>Soy un parrafo</p>
        // </Fragment>
    );

  
    
}

// con PropTypes establecemos propiedades necesarias para los props que recibe un componente 

PrimeraApp.propTypes = {
    // acá le decimos que saludo es requerido y que debe ser de tipo string 
    saludo: PropTypes.string.isRequired
}

export default PrimeraApp;