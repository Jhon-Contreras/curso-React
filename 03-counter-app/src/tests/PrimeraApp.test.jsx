import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import PrimeraApp from "../PrimeraApp";
import { shallow } from 'enzyme';

describe('Pruebas en <PrimeraApp />', () => {
//prueba con  import '@testing-library/jest-dom/extend-expect'; en setupTests.js

    //   test('Debe de mostrar el mensaje "hola mundo"', () => {
    
//     const saludo = 'hola mundo';
//     const { getByText } = render(<PrimeraApp saludo={saludo}/>);

//     expect(getByText(saludo)).toBeInTheDocument();

//   });

    test('Debe de mostrar <PrimeraApp /> correctamente', () => {
      const saludo = 'hola mundo';
      const wrapper = shallow(<PrimeraApp saludo={ saludo }/>); //shallow es parecido al render()

      expect( wrapper ).toMatchSnapshot();
    });
    test('debe de mostrar el subtitulo enviado por props', () => {
      const saludo = 'hola mundo';
      const subTitulo = 'Soy un parrafo desde defaultProps';

      const wrapper = shallow( <PrimeraApp saludo={ saludo } subTitulo={subTitulo} /> );

      const textoParrafo = wrapper.find('p').text();
     
      expect(textoParrafo).toBe(subTitulo);
    });
    
    
});


