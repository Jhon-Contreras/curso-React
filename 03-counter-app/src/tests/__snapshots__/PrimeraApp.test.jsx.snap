// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    hola mundo
  </h1>
  <p>
    Soy un parrafo desde defaultProps
  </p>
</Fragment>
`;
